Rationale.md

```diff
@@ -0,0 +1,215 @@
+# Design Rationale
+
+WebAssembly was designed incrementally, and desisions were made with a solid
+rationale which may turn out to be incorrect in the future. Sometimes, decisions
+were made without the luxury of data, and the designers would like to revisit
+certain decisions once real end-to-end usecases are met and Science can be
+performed.
+
+You, the reader, may also be interested in understanding why
+[Abstract Syntax Tree (AST)](AstSemantics.md) semantics are the way they are,
+and digging through github's issues and pull requests becomes difficult.
```

Comment:
Are the above two paragraphs necessary? It's a rationale document; it's own purpose should be self-evident. And, I don't fully agree with the characterization of the process given here.